src/eip7594/eip7594.c

```diff
@@ -454,6 +454,125 @@ static C_KZG_RET compute_r_powers_for_verify_cell_kzg_proof_batch(
     return ret;
 }
 
+/**
+ * Aggregate columns, compute the sum of interpolation polynomials, and commit to the result.
+ *
+ * This function computes: [sum_k r^k interpolation_poly_k(s)]
+ *
+ * This function first aggregates cells from the same column by scaling them with the corresponding
+ * powers of the random challenge. It then computes the sum of the interpolation polynomials from
+ * the aggregated columns and finally commits to the aggregated interpolation polynomial.
+ *
+ * @param[out]  commitment_out              Commitment to the aggregated interpolation poly
+ * @param[in]   r_powers                    Precomputed powers of the random challenge
+ * @param[in]   cell_indices                Indices of the cells
+ * @param[in]   cells                       Array of cells
+ * @param[in]   num_cells                   Number of cells
+ * @param[in]   s                           The trusted setup
+ */
+static C_KZG_RET compute_commitment_to_aggregated_interpolation_poly(
+    g1_t *commitment_out,
+    const fr_t *r_powers,
+    const uint64_t *cell_indices,
+    const Cell *cells,
+    size_t num_cells,
+    const KZGSettings *s
+) {
+    C_KZG_RET ret;
+    fr_t *aggregated_column_cells = NULL;
+    fr_t *column_interpolation_poly = NULL;
+    fr_t *aggregated_interpolation_poly = NULL;
+    bool *is_cell_used = NULL;
+
+    // Allocate memory
+    ret = new_fr_array(&aggregated_column_cells, FIELD_ELEMENTS_PER_EXT_BLOB);
+    if (ret != C_KZG_OK) goto out;
+    ret = new_fr_array(&column_interpolation_poly, FIELD_ELEMENTS_PER_CELL);
+    if (ret != C_KZG_OK) goto out;
+    ret = new_fr_array(&aggregated_interpolation_poly, FIELD_ELEMENTS_PER_CELL);
+    if (ret != C_KZG_OK) goto out;
+    ret = new_bool_array(&is_cell_used, FIELD_ELEMENTS_PER_EXT_BLOB);
+    if (ret != C_KZG_OK) goto out;
+
+    // Initialize aggregated columns and usage flags
+    for (size_t i = 0; i < CELLS_PER_EXT_BLOB; i++) {
+        for (size_t j = 0; j < FIELD_ELEMENTS_PER_CELL; j++) {
+            size_t index = i * FIELD_ELEMENTS_PER_CELL + j;
+            aggregated_column_cells[index] = FR_ZERO;
+            is_cell_used[index] = false;
+        }
+    }
```

Comment:
Hmm "flags" isn't the ideal word here. Not sure what is though. Maybe "markers"?